sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java

```diff
@@ -274,6 +274,7 @@ public void setup() {
 
     // Futures ignored as exceptions will be flushed out in the commitTxn
     @SuppressWarnings("FutureReturnValueIgnored")
+    @RequiresStableInput
```

Comment:
@kennknowles will this have any effect on Dataflow / DirectRunner? E.g. it could introduce extra fusion break/shuffle. There is an explicit GBK between `Sequencer` and `ExactlyOnceWriter`. My understanding is that `@RequiresStableInput` makes that unnecessary.  
@mxm, might be better to leave comment about remove it.

Reply:
Right now it does nothing. Also when we implement RequiresStableInput for the Dataflow runner, we should probably make it a noop in these cases (since we know the input will already be stable).

Reply:
@reuvenlax FYI, `@RequiresStableInput` is already supported on Dataflow runner (PR: https://github.com/apache/beam/pull/5787), so I believe it will add an extra `Reshuffle` instead of doing nothing.

In my previous work, I didn't replace the hard-coded `Reshuffle`s with `@RequiresStableInput`, mainly because we are not sure how it will affect Spark and Flink runners.